agent/api/task.go

```diff
@@ -186,13 +186,22 @@ func TaskFromACS(acsTask *ecsacs.Task, envelope *ecsacs.PayloadMessage) (*Task,
 // PostUnmarshalTask is run after a task has been unmarshalled, but before it has been
 // run. It is possible it will be subsequently called after that and should be
 // able to handle such an occurrence appropriately (e.g. behave idempotently).
-func (task *Task) PostUnmarshalTask(cfg *config.Config, credentialsManager credentials.Manager) {
+func (task *Task) PostUnmarshalTask(cfg *config.Config,
+	credentialsManager credentials.Manager, resourceFields *taskresource.ResourceFields) error {
 	// TODO, add rudimentary plugin support and call any plugins that want to
 	// hook into this
 	task.adjustForPlatform(cfg)
+	if task.MemoryCPULimitsEnabled {
+		err := task.initializeCgroupResource(cfg.CgroupPath, resourceFields)
+		if err != nil {
+			seelog.Errorf("Task [%s]: could not intialize resource: %v", task.Arn, err)
+			return apierrors.ResourceInitializeError{task.Arn}
```

Comment:
We're loosing context of the original error here. Can you please use `errors.Wrapf()` or make sure that `ResourceInitializeError` has an `origError` or some other similar field so that we preserve it?